src/pointer-analysis/value_set.h

```diff
@@ -317,6 +317,15 @@ class value_sett
     values.clear();
   }
 
+  /// Finds an entry in this value-set. The interface differs from get_entry
+  /// because get_value_set_rec wants to check for a struct's first component
+  /// before stripping the suffix as is done in get_entry.
+  /// \param id: identifier to find.
+  /// \return a constant pointer to an entry if found, or null otherwise.
+  ///   Note the pointer may be invalidated by insert operations, including
+  ///   get_entry.
+  const entryt *find_entry(const idt &id) const;
```

Comment:
What about `optionalt<T *>`? It would more clearly express the intent and remove the role of `nullptr` as a sentinel value that can accidentally be dereferenced.